### PR TITLE
[yaml] Add tests for provider schema validation 

### PR DESCRIPTION
**Pull request by Polber**
Add tests to validate #31974.

Created separate PR due to out-of-scope changes that should avoid cherry pick into 2.58.0 release candidate.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
```
ERROR: /runner/_work/beam/beam/sdks/python/test-suites/tox/pycommon/build/srcs/sdks/python/apache_beam/yaml/yaml_transform_test.py Imports are incorrectly sorted.
--- /runner/_work/beam/beam/sdks/python/test-suites/tox/pycommon/build/srcs/sdks/python/apache_beam/yaml/yaml_transform_test.py:before	2024-07-24 23:50:37.354358
+++ /runner/_work/beam/beam/sdks/python/test-suites/tox/pycommon/build/srcs/sdks/python/apache_beam/yaml/yaml_transform_test.py:after	2024-07-24 23:59:06.838165
@@ -23,16 +23,15 @@
 import unittest
 
 import yaml
-from apache_beam.yaml.yaml_transform import SafeLineLoader
-
-from apache_beam.yaml.yaml_transform import expand_pipeline
 
 import apache_beam as beam
 from apache_beam.testing.util import assert_that
 from apache_beam.testing.util import equal_to
 from apache_beam.utils import python_callable
 from apache_beam.yaml import yaml_provider
+from apache_beam.yaml.yaml_transform import SafeLineLoader
 from apache_beam.yaml.yaml_transform import YamlTransform
+from apache_beam.yaml.yaml_transform import expand_pipeline
 
```

File: sdks/python/apache_beam/yaml/pipeline.schema.yaml [hunk 1]
```diff
@@ -85,7 +85,10 @@ $defs:
         items:
           properties:
             # Must be implicit.
-            input: { not: {} }
+            input:
+              oneOf:
+                - { "not": {} }
+                - { "enum": [ "__explicitly_empty__" ] }
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/pipeline.schema.yaml:91]: It feels like we shouldn't be validating once we've injected the `__explicitly_empty__` strings.

File: sdks/python/apache_beam/yaml/yaml_transform.py [hunk 2]
```diff
@@ -115,6 +115,10 @@ def explicitly_empty():
 
 
 def is_explicitly_empty(io):
+  if isinstance(io, str):
```
Comment (robertwb) [inline sdks/python/apache_beam/yaml/yaml_transform.py:118]: Likewise, inputs should have been normalized before we hit this. is_explicitly_empty should only process normalized inputs, not raw strings.

**Comment by damccorm** [comment]
waiting on author

**Pull request closed without merging**